addOns/graphql/src/main/java/org/zaproxy/addon/graphql/ImportFromFileDialog.java

```diff
@@ -0,0 +1,97 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.graphql;
+
+import java.awt.GridBagConstraints;
+import java.io.IOException;
+import javax.swing.JButton;
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.view.View;
+
+public class ImportFromFileDialog extends ImportFromAbstractDialog {
+
+    private static final long serialVersionUID = 1L;
+    private static final String MESSAGE_PREFIX = "graphql.importfromfiledialog.";
+
+    private JButton buttonChooseFile;
+
+    public ImportFromFileDialog(JFrame parent, ExtensionGraphQl caller) {
+        super(
+                parent,
+                caller,
+                Constant.messages.getString(MESSAGE_PREFIX + "title"),
+                Constant.messages.getString(MESSAGE_PREFIX + "labelfile"));
+    }
+
+    @Override
+    protected void addFromFields(GridBagConstraints constraints) {
+        constraints.gridwidth = 2;
+        super.addFromFields(constraints);
+
+        constraints.gridx = 3;
+        constraints.gridwidth = 1;
+        add(getButtonChooseFile(), constraints);
+    }
+
+    private JButton getButtonChooseFile() {
+        if (buttonChooseFile == null) {
+            buttonChooseFile =
+                    new JButton(Constant.messages.getString(MESSAGE_PREFIX + "choosefilebutton"));
+
+            buttonChooseFile.addActionListener(
+                    e -> {
+                        JFileChooser filechooser =
+                                new JFileChooser(
+                                        Model.getSingleton().getOptionsParam().getUserDirectory());
+                        int state = filechooser.showOpenDialog(View.getSingleton().getMainFrame());
+                        if (state == JFileChooser.APPROVE_OPTION) {
+                            try {
+                                getFromField()
+                                        .setText(filechooser.getSelectedFile().getCanonicalPath());
+                                Model.getSingleton()
+                                        .getOptionsParam()
+                                        .setUserDirectory(filechooser.getCurrentDirectory());
+                            } catch (IOException ex) {
+                                showWarningDialog(
+                                        Constant.messages.getString(MESSAGE_PREFIX + "badfile"));
+                            }
+                        }
+                    });
+        }
+        return buttonChooseFile;
+    }
+
+    @Override
+    protected boolean importDefinition() {
+        // Import and parse schema if file path is provided
+        if (!getFromField().getText().isEmpty()) {
+            // TO DO
```

Comment:
Is this still a TODO? If so some more details might help you remember whats TODO in the future ;)

Reply:
Yes it is. Updated :).